### PR TITLE
Sync ignore rapid create delete

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
## Submitter Checklist:

Fixes https://github.com/brave/brave-browser/issues/2449 .

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
Manual: go through
Auto: `npm run test -- brave_unit_tests --filter=BraveBookmarkChangeProcessorTest.IgnoreRapidCreateDelete`.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_sync/client/bookmark_change_processor_unittest.cc [hunk 1]
```diff
@@ -1165,3 +1165,16 @@ TEST_F(BraveBookmarkChangeProcessorTest, ItemAheadOfFolderRequireStrictSorting)
...
 }
+
+TEST_F(BraveBookmarkChangeProcessorTest, IgnoreRapidCreateDelete) {
+  change_processor()->Start();
+
+  const auto* node_a = model()->AddURL(model()->other_node(), 0,
+                           base::ASCIIToUTF16("A.com - title"),
+                           GURL("https://a.com/"));
+  model()->Remove(node_a);
+
```
[the remark this answers is not in the document]
Comment (AlexeyBarabash) [reply components/brave_sync/client/bookmark_change_processor_unittest.cc]: Thanks, modified this.

**Comment by AlexeyBarabash** [comment]
Added check for child count == 0 at `Deleted Bookamrks` at the unit test.
https://github.com/brave/brave-core/pull/1061/commits/035eff7eaec5d51449fb07ba0ad4ff109eade7a1#diff-4c9d224d445d04212cbe16c50549cd4cR1177

**Comment by AlexeyBarabash** [comment]
master: https://github.com/brave/brave-core/commit/bd653cb74486b49325b392172deb9659787cc981
0.60.x: https://github.com/brave/brave-core/commit/cb737b03c5ab0d6145f11b3e244cf3dd05e57f65

**Comment by rebron** [comment]
Approving uplift-request for 0.59.x-beta. cc: @kjozwiak cc: @srirambv

**Comment by darkdh** [comment]
0.59.x: 84a58918b87d796e432e0071f8509c6cfa2e8766

**Pull request merged**